File: client/FieldInfo.py
from card import Card

class FieldInfo:
    def __init__(
        self,
        start_flag  : bool,       # 游戏是否开始
        is_player   : bool,       # 当前用户是否是玩家

        # 玩家信息
        client_id   : int,        # 当前玩家的ID
        client_cards: list[Card], # 当前玩家的手牌

        # 场面信息
        user_names        : list[str],        # 所有玩家的名字
        user_scores       : list[int],        # 所有玩家的分数
        users_cards_num   : list[int],        # 所有玩家的牌数
        users_cards       : list[list[Card]], # 所有玩家的牌
        users_played_cards: list[list[Card]], # 所有玩家的出牌
        head_master       : int,              # 头科
        now_score         : int,              # 场面分数
        now_player        : int,              # 当前出牌的玩家
        last_player       : int,              # 上一出牌的玩家
        his_now_score     : int,              # 历史场上分数
        his_last_player   : int,              # 历史上一出牌的玩家
    ):
        self.start_flag = start_flag
        self.is_player = is_player
        self.client_id = client_id
        self.client_cards = client_cards
        self.user_names = user_names
        self.user_scores = user_scores
        self.users_cards_num = users_cards_num
        self.users_cards = users_cards
        self.users_played_cards = users_played_cards
        self.head_master = head_master
        self.now_score = now_score
        self.now_player = now_player
        self.last_player = last_player
        self.his_now_score = his_now_score
        self.his_last_player = his_last_player

File: client_dart/lib/core/models/field_info.dart
import 'package:liujiatong/core/models/card.dart';

/// 牌局场面信息 DTO（与 client 使用的场面信息一致）
class FieldInfo {
  FieldInfo({
    required this.startFlag,
    required this.isPlayer,
    required this.clientId,
    required this.clientCards,
    required this.userNames,
    required this.userScores,
    required this.usersCardsNum,
    required this.usersCards,
    required this.usersPlayedCards,
    required this.headMaster,
    required this.nowScore,
    required this.nowPlayer,
    required this.lastPlayer,
    required this.hisNowScore,
    required this.hisLastPlayer,
  });

  final bool startFlag;
  final bool isPlayer;
  final int clientId;
  final List<Card> clientCards;
  final List<String> userNames;
  final List<int> userScores;
  final List<int> usersCardsNum;
  final List<List<Card>> usersCards;
  final List<List<Card>> usersPlayedCards;
  final int headMaster;
  final int nowScore;
  final int nowPlayer;
  final int lastPlayer;
  final int hisNowScore;
  final int? hisLastPlayer;
}
